libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -561,6 +553,68 @@ inline _CUDA_VSTD::uint64_t * barrier_native_handle(barrier<thread_scope_block>
     return reinterpret_cast<_CUDA_VSTD::uint64_t *>(&b.__barrier);
 }
 
+
+_LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_DEVICE
+inline barrier<thread_scope_block>::arrival_token arrive_tx(
+    barrier<thread_scope_block> & __b,
+    _CUDA_VSTD::ptrdiff_t __arrive_count_update,
+    _CUDA_VSTD::ptrdiff_t __transaction_count_update) {
+    _LIBCUDACXX_DEBUG_ASSERT(__arrive_count_update >= 0, "Arrival count update must be non-negative.");
+    _LIBCUDACXX_DEBUG_ASSERT(__transaction_count_update >= 0, "Transaction count update must be non-negative.");
+    // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#contents-of-the-mbarrier-object
+    _LIBCUDACXX_DEBUG_ASSERT(__transaction_count_update <= (1 << 20) - 1, "Transaction count update cannot exceed 2^20 - 1.");
+    barrier<thread_scope_block>::arrival_token __token = {};
```

Comment:
```suggestion
    typename barrier<thread_scope_block, _CUDA_VSTD::__empty_completion>::arrival_token __token = {};
```

Reply:
Drop this suggested change, this function should not be a template on the completion function, because we only do mbarrier when the completion function is the default one.

Reply:
Resolving this discussion in lieu of the many other places where this is discussed.